src/configure/clients/modaTemplateServiceClient.ts

```diff
@@ -0,0 +1,87 @@
+
+import { RepositoryAnalysis } from "azureintegration-repoanalysis-client-internal";
+import { RestClient } from "typed-rest-client";
+import vscodeUri from "vscode-uri";
+import { ExtendedPipelineTemplate } from "../model/Contracts";
+import { StringMap } from "../model/models";
+import { TemplateInfo } from "../model/templateModels";
+import { ITemplateServiceClient } from "./ITemplateServiceClient";
+
+export class ModaTemplateServiceClient implements ITemplateServiceClient {
+    private restClient: RestClient;
+    private githubPat: string;
+    private pathUrl: string;
+    private readonly apiVersion = "6.0-preview.1";
+    private readonly extendedPipelineTemplateResource = "ExtendedPipelineTemplates";
+    private readonly templatesInfoResource = "TemplatesInfo";
+    private readonly templateAssetFilesResource = "TemplateAssetFiles";
+
+    constructor(url: string, githubPat: string) {
+        const u = vscodeUri.parse(url);
+        this.restClient = new RestClient("deploy-to-azure", u.scheme + "://" + u.authority);
+        this.pathUrl = u.path;
+        this.githubPat = githubPat;
+    }
+
+    public async getTemplates(body: RepositoryAnalysis): Promise<TemplateInfo[]> {
+        const requestUri = this.pathUrl + this.templatesInfoResource;
```

Comment:
why do we need to completely re-write the full function in both clients. Can't we just set url and auth header separately and then keep the rest of the code common?